rclcpp/src/rclcpp/clock.cpp

```diff
@@ -79,8 +82,20 @@ Clock::now() const
   return now;
 }
 
+void
+Clock::cancel_sleep_or_wait()
+{
+  {
+    std::unique_lock lock(impl_->wait_mutex_);
+    impl_->stop_sleeping_ = true;
+  }
+  impl_->cv_.notify_one();
+}
+
 bool
-Clock::sleep_until(Time until, Context::SharedPtr context)
+Clock::sleep_until(
```

Comment:
is it possible for multiple threads to call `sleep_until` on the same clock object?

Reply:
Yes one could do that. The cancel mechanism will only wake up one thread currently.

Reply:
I just checked the documentation of clock. It does not explicitly state that it is safe for usage from multiple threads, therefore I would not assume that it needs to handle this case.